packages/renderer/src/lib/image/ImageDetailsCheck.svelte

```diff
@@ -0,0 +1,149 @@
+<script lang="ts">
+import type { ImageCheck } from '@podman-desktop/api';
+import type { ImageCheckerInfo } from '../../../../main/src/plugin/api/image-checker-info';
+import type { ImageInfoUI } from './ImageInfoUI';
+import Markdown from '../markdown/Markdown.svelte';
+import Fa from 'svelte-fa';
+import { faCheckCircle, faExclamationCircle, faStethoscope } from '@fortawesome/free-solid-svg-icons';
+import Button from '../ui/Button.svelte';
+import WarningMessage from '../ui/WarningMessage.svelte';
+import type { Unsubscriber } from 'svelte/store';
+import { onDestroy, onMount } from 'svelte';
+import { imageCheckerProviders } from '/@/stores/image-checker-providers';
+
+export let image: ImageInfoUI;
+
+interface CheckUI {
+  provider: ImageCheckerInfo;
+  check: ImageCheck;
+}
+
+let providers: ImageCheckerInfo[];
+let results: CheckUI[] = [];
+let cancellableTokenId: number = 0;
+
+let remainingProviders: number;
+let aborted = false;
+
+let providersUnsubscribe: Unsubscriber;
+
+onMount(async () => {
+  providersUnsubscribe = imageCheckerProviders.subscribe(providers => {
+    callProviders(providers);
+  });
+});
+
+onDestroy(() => {
+  // unsubscribe from the store
+  if (providersUnsubscribe) {
+    providersUnsubscribe();
+  }
+});
+
+async function callProviders(_providers: readonly ImageCheckerInfo[]) {
```

Comment:
I'm wondering why the aggregation is done on the UI side (which needs more exchanges with the main process) and why it's not done in the main process (single call)

Reply:
It helps differentiate the results of each provider (the user can Abort before waiting all providers to return results, and it will get the results of the first providers).